### DISCUSSION TITLE
How can I get "request URL" data from the request?

### DISCUSSION THREAD

**Discussion by MaxAllianov** [cypress-io/cypress#20006] [Questions and Help] [answered]
How can I get the "request URL" value from the 301 request?
Cypress ignore the redirect for some reason, so I want to get the request URL and follow it manually via the Visit. 
But unfortunately, I don't know how to get this data.

![image](https://user-images.githubusercontent.com/76516974/152059101-cfca7b52-1cd3-4ae8-ad1b-a93a12a15455.png)

Please help me to reach the 'request URL' from the request!

**Comment by MaxAllianov** [comment]
I have tried to do this, but unfortunately, console.log and visit are skipped:
```
cy.intercept('GET', (my request), (req) => {
      const reqUrl = req.url
      console.log(reqUrl)
      cy.visit(reqUrl)
    })
```

**Reply by jjhelguero** [reply]
What did you input as your `(my request)`?

**Reply by MaxAllianov** [reply]
> What did you input as your `(my request)`?

I have tried to put regex and {path}.
The (my request) is 100% correct (I tried to alias it and wait, it went thru successfully)

Lets say the request is 
**website.com/*******?payment_success**
(where ******* are 7 random numbers)
In this case my request would be:
```
cy.intercept('GET', /website\.com\/.*\?payment_success/ , (req) => {
      const reqUrl = req.url
      console.log(reqUrl)
      cy.visit(reqUrl)
    })
```

**Comment by MaxAllianov** [comment, marked as answer]
For those who are stuck on this like me... :)
I have found a solution that met my needs:

If the request is: **website.com/1234567?payment_success**
then the code is:

```
cy.intercept(/website\.com\/.*\?payment_success/).as('myAlias')
cy.wait('@myAlias').then((interception) => {
cy.visit(interception.request.url)
})
```

Here are relevant doc:
https://docs.cypress.io/api/commands/intercept#Using-the-yielded-object

The thread could be closed

**Reply by jjhelguero** [reply]
Nice. I'd recommend updating regex to `/website.com\/(\d{7})\?payment_success/` to ensure there are 7 random numbers included in your link.
